.circleci/config.yml

```diff
@@ -95,33 +202,46 @@ commands:
           command: |
             npm config set scripts-prepend-node-path true
             npm install
-            npm install -g sfdx-cli@7.45.1
+            npm install -g sfdx-cli@<< pipeline.parameters.cli-version >>
             npm install -g codecov
             npm install -g junit-merge
       - run:
           name: 'Compile'
           shell: bash.exe
           command: npm run compile
-      - save_cache:
-          key: dependency-cache-win3-{{ checksum "package.json" }}
-          paths:
-            - node_modules
+      - save_dependency_cache:
+          base-cache-path: 'dependency-cache-win3-'
       - run:
           name: 'Run lint'
           shell: bash.exe
           command: |
             npm run lint
 
+executors:
+  default-executor:
+    parameters:
+      # We are unable to use the recommended cimg docker images since they use the root path instead of home
+      image:
+        type: string
+        default: circleci/node:<< pipeline.parameters.node-version >>
+      working-dir:
+        type: string
+        default: ~/cli
+    docker:
+      - image: << parameters.image >>
+    working_directory: << parameters.working-dir >>
+
 jobs:
   build-all:
     description: 'Build artifacts in a non-Windows environment'
-    docker:
-      - image: circleci/node:12.4.0
+    executor:
+      name: default-executor
+      working-dir: ~/project
     environment:
       - SFDX_URL_LINUX: https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz
```

Comment:
I opted to keep a lot of our environment variables as is instead of changing them to pipeline parameters. If anyone has any suggestions on ones that would be good to easily control via the pipeline please feel free to make them.